src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.NodeMapBuilder.cs

```diff
@@ -257,12 +257,28 @@ public override BoundNode VisitBinaryOperator(BoundBinaryOperator node)
 
             public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator node)
             {
-                VisitList(node.LeftVariables);
+                Visit(node.Left);
                 Visit(node.Right);
                 // don't map the deconstruction, conversion or assignment steps
                 return null;
             }
 
+            public override BoundNode VisitForEachStatement(BoundForEachStatement node)
```

Comment:
It feels like it shouldn't be skipped because it contains bound nodes associated with syntax that isn't covered by bound nodes outside of the DeconstructionOpt.